pkg/kbagent/service/command.go

```diff
@@ -66,36 +65,31 @@ func runCommand(ctx context.Context, action *proto.ExecAction, parameters map[st
 		}
 		return nil, err
 	}
-	return <-stdoutChan, nil
+	return outBuffer.Bytes(), nil
 }
 
-func runCommandNonBlocking(ctx context.Context, action *proto.ExecAction, parameters map[string]string, timeout *int32) (chan []byte, chan []byte, chan error, error) {
+func runCommandNonBlocking(ctx context.Context, action *proto.ExecAction, parameters map[string]string, timeout *int32) (*bytes.Buffer, *bytes.Buffer, chan error, error) {
```

Comment:
Directly returning *bytes.Buffer doesn't seem like a good idea. The caller might use the buffer before the channel returns, potentially causing concurrency issues. I prefer returning a struct through the channel, which contains these two buffers and the execution result of the command (i.e., err error).

```golang
type commandResult struct {
  err error
  stdout *bytes.Buffer
  stderr *bytes.Buffer
}

error, error) {
func runCommandNonBlocking(ctx context.Context, action *proto.ExecAction, parameters map[string]string, timeout *int32) (chan *commandResult, error) {
  // ...
}
```

@leon-inf What do you think?

Reply:
Both are good to me, and this method is safer and less error-prone.